warn/scrapers/ga.py

```diff
@@ -1,121 +1,108 @@
-import re
 from datetime import datetime
 from pathlib import Path
 
-from bs4 import BeautifulSoup
+from spatula import CSS, HtmlListPage, ListPage, NullSource, SkipItem
 
 from .. import utils
-from ..cache import Cache
+from ..cache import CachedURL
 
 __authors__ = ["chriszs"]
 __tags__ = ["html"]
 
 
-def scrape(
-    data_dir: Path = utils.WARN_DATA_DIR,
-    cache_dir: Path = utils.WARN_CACHE_DIR,
-) -> Path:
-    """
-    Scrape data from Georgia.
+class YearList(ListPage):
+    """The list of years to scrape."""
 
-    Keyword arguments:
-    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
-    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+    def __init__(self, *, start, end=None, **kwargs):
+        """Initialize a new instance."""
+        super().__init__(None, **kwargs)
+        self.start = start
+        self.end = end or datetime.now().year
 
-    Returns: the Path where the file is written
-    """
-    state_code = "ga"
-    cache = Cache(cache_dir)
+    @property
+    def years(self):
+        """Get the list of years to scrape."""
+        return reversed(range(self.start, self.end + 1))
```

Comment:
I really want to factor the year range part of this out from the `NoticeList` generation. Still working on how to do that.